File: src/main/java/project/models/Service.java
package project.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import project.saloon.Saloon;

import javax.persistence.*;
import java.io.Serializable;
import java.util.Set;

// Service serves to list the services of the various saloons, one saloons has many services

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "service")
public class Service implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private String name;
    private Double price;
    private String description;
    private String available;


    // Salão que oferece o serviço
    @ManyToOne
    @JoinColumn(name="saloon_id")
    private Saloon saloon;

    // Horario do serviço
    @OneToMany(mappedBy="service")
    private Set<Schedule> schedules;


    // reservas do serviço
    @OneToMany(mappedBy="services")
    private Set<Reservation> reservation;



}

File: mobile_app/lib/models/Service.dart
import 'package:mobile_app/models/Saloon.dart';


class Service{
  String name;
  double price;
  String description;
  String available;
  Saloon saloon;
  int id;



  Service();

  Service.fromMap(Map<String, dynamic> data){
    id=data['id'];
    name=data['name'];
    price=data['price'];
    description=data['description'];
    available=data['available'];
    saloon=Saloon.fromMap(data['saloon']);
  
  }

  Map<String, dynamic> toMap() {
    return {
      'id':id,
     'name':name,
    'price':price,
    'description':description,
    'available':available,
    'saloon':saloon.toMap()
    };
  }

}
